lib/logstash/inputs/kafka.rb

```diff
@@ -98,7 +104,7 @@ def register
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
     @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = Kafka::Group.new(options)
+    @consumer_group = @kafka_group_class.new(options)
```

Comment:
If there's a need for multiple consumers then accepting the consumer class as an argument makes sense indeed. 

If not, then I'd probably consider a more functional oriented refactoring:

```diff
diff --git a/lib/logstash/inputs/kafka.rb b/lib/logstash/inputs/kafka.rb
index a417ee9..e1b1210 100644
--- a/lib/logstash/inputs/kafka.rb
+++ b/lib/logstash/inputs/kafka.rb
@@ -80,12 +80,6 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   config :fetch_message_max_bytes, :validate => :number, :default => 1048576
 
   public
-  def initialize(params={}, kafka_group_class=Kafka::Group)
-    super(params)
-    @kafka_group_class = kafka_group_class
-  end
-
-  public
   def register
     require 'jruby-kafka'
     options = {
@@ -104,7 +98,7 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
     @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = @kafka_group_class.new(options)
+    @consumer_group = create_consumer_group(options)
     @logger.info('Registering kafka', :group_id => @group_id, :topic_id => @topic_id, :zk_connect => @zk_connect)
   end # def register
 
@@ -156,4 +150,8 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
     end # begin
   end # def queue_event
 
+  private
+  def create_consumer_group(options)
+    Kafka::Group.new(options)
+  end
 end #class LogStash::Inputs::Kafka
diff --git a/spec/inputs/kafka_spec.rb b/spec/inputs/kafka_spec.rb
index f1d2ea4..4cfe0b8 100644
--- a/spec/inputs/kafka_spec.rb
+++ b/spec/inputs/kafka_spec.rb
@@ -3,6 +3,12 @@ require "logstash/devutils/rspec/spec_helper"
 require "logstash/inputs/kafka"
 require 'jruby-kafka'
 
+class TestKafkaGroup < Kafka::Group
+  def run(a_num_threads, a_queue)
+    a_queue << 'Kafka message'
+  end
+end
+
 class LogStash::Inputs::TestKafka < LogStash::Inputs::Kafka
   milestone 1
   private
@@ -11,12 +17,10 @@ class LogStash::Inputs::TestKafka < LogStash::Inputs::Kafka
     # need to raise exception here to stop the infinite loop
     raise LogStash::ShutdownSignal
   end
-end
 
-
-class TestKafkaGroup < Kafka::Group
-  def run(a_num_threads, a_queue)
-    a_queue << 'Kafka message'
+  private
+  def create_consumer_group(options)
+    TestKafkaGroup.new(options)
   end
 end
 
@@ -38,7 +42,7 @@ describe 'inputs/kafka' do
   end
 
   it 'should retrieve event from kafka' do
-    kafka = LogStash::Inputs::TestKafka.new(kafka_config, TestKafkaGroup)
+    kafka = LogStash::Inputs::TestKafka.new(kafka_config)
     kafka.register
 
     logstash_queue = Queue.new
@@ -50,7 +54,7 @@ describe 'inputs/kafka' do
   end
 
   it 'should retrieve a decorated event from kafka' do
-    kafka = LogStash::Inputs::TestKafka.new(decorated_kafka_config, TestKafkaGroup)
+    kafka = LogStash::Inputs::TestKafka.new(decorated_kafka_config)
     kafka.register
 
     logstash_queue = Queue.new
```